File: backend/padel/src/main/java/be/ephec/pdw/padel/model/Match.java
package be.ephec.pdw.padel.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;

@Entity
@Table(name = "match_padel") // mot sql
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Match {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDateTime dateHeureDebut;

    private boolean estPublic;

    @ManyToOne
    @JoinColumn (name = "terrain_id")
    private Terrain terrain;

    @ManyToOne
    @JoinColumn(name = "organisateur_matricule")
    private Membre organisateur;

    @Enumerated(EnumType.STRING)
    private StatutMatch statut;

    @OneToMany(mappedBy = "match")
    @JsonManagedReference
    private List<Reservation> reservations;

}

File: frontend/src/app/core/models/match.ts
export type StatutMatch = 'PLANIFIE' | 'TERMINE' | 'ANNULE' | 'COMPLET';

export interface Match {
  id: number;
  nbParticipants: number;
  terrainNom?: string;
  terrain?: string;
  siteId?: number;
  site?: string;
  dateHeureDebut: string;
  organisateurMatricule: string;
  estPublic: boolean;
  statut: StatutMatch;
}

export interface CreateMatchPayload {
  organisateur_matricule?: string;
  terrainID: number;
  date: string;
  estPublic: boolean;
}
